squire_web/src/tournament/pairings/mod.rs

```diff
@@ -74,108 +81,118 @@ pub enum PairingsViewMode {
-        let mut to_return = Self {
-            id,
-            admin_id,
-            names: None,
-            send_names: ctx.link().callback(PairingsViewMessage::PlayerNamesReady),
+    fn v_create(ctx: &Context<TournViewerComponentWrapper<Self>>, _state: &WrapperState) -> Self {
+        let to_return = Self {
+            id: ctx.props().t_id,
             mode: PairingsViewMode::CreatePairings,
-            pairings: None,
-            send_pairings: ctx.link().callback(PairingsViewMessage::PairingsReady),
-            active: None,
-            send_active: ctx.link().callback(PairingsViewMessage::ActiveRoundsReady),
-            max_player_count: None,
-            send_max_player_count: ctx.link().callback(PairingsViewMessage::MatchSizeReady),
             single_round_inputs: Vec::new(),
             single_bye_input: "".to_string(),
-            send_op_result,
+            query_data: None,
+            pairings: None,
         };
-        to_return.query_player_names(ctx);
-        to_return.query_match_size(ctx);
         to_return
     }
 
-    fn update(&mut self, ctx: &Context<Self>, msg: Self::Message) -> bool {
-        match msg {
-            PairingsViewMessage::PlayerNamesReady(pnhm) => {
-                self.names = Some(pnhm);
-                true
+    fn v_view(
+        &self,
+        ctx: &Context<TournViewerComponentWrapper<Self>>,
+        _state: &WrapperState,
+    ) -> Html {
+        let make_callback = |mode| {
+            ctx.link().callback(move |_| {
+                WrapperMessage::Interaction(PairingsViewMessage::ChangeMode(mode))
+            })
+        };
+        let make_button =
+            |name, mode| html! { <button onclick = { make_callback(mode) }>{name}</button> };
+        html! {
+            <div>
+                <h1>{ "Pairings" }</h1>
+                <div>
+                    <>{ make_button("Create Pairings", PairingsViewMode::CreatePairings) }</>
+                    <>{ make_button("Active Pairings", PairingsViewMode::ActivePairings) }</>
+                    <>{ make_button("Create single matches", PairingsViewMode::CreateSingleMatches) }</>
+                </div>
+                <div>{
+                    match self.mode {
+                        PairingsViewMode::CreatePairings => {
+                            self.view_creation_menu(ctx)
+                        }
+                        PairingsViewMode::ActivePairings => {
+                            self.view_active_menu(ctx)
+                        }
+                        PairingsViewMode::CreateSingleMatches => {
+                            self.view_single_menu(ctx)
+                        }
+                    }
+                }</div>
+            </div>
+        }
+    }
+
+    fn load_queried_data(&mut self, _msg: Self::QueryMessage, _state: &WrapperState) -> bool {
+        match _msg {
+            PairingsQueryMessage::AllDataReady(_data) => {
+                self.query_data = Some(_data);
```

Comment:
Same here with `_data`.
```suggestion
            PairingsQueryMessage::AllDataReady(data) => {
                self.query_data = Some(data);
```